weave-js/src/components/FancyPage/useProjectSidebar.ts

```diff
@@ -9,13 +9,20 @@ export const useProjectSidebar = (
   viewingRestricted: boolean,
   hasModelsData: boolean,
   hasWeaveData: boolean,
-  isLargeWorkspaceModeEnabled: boolean
+  isLargeWorkspaceModeEnabled: boolean,
+  hasTraceBackend: boolean = true
 ): FancyPageSidebarItem[] => {
-  const isNoData = !hasModelsData && !hasWeaveData;
-  const isModelsOnly = hasModelsData && !hasWeaveData;
-  const isWeaveOnly = !hasModelsData && hasWeaveData;
-  const isBothData = hasModelsData && hasWeaveData;
-  const isShowAll = isNoData || isBothData;
+  // Should show models sidebar items if we have models data or if we don't have a trace backend
+  const showModelsSidebarItems = hasModelsData || !hasTraceBackend;
+  // Should show weave sidebar items if we have weave data and we have a trace backend
```

Comment:
Is is possible to have weave data and not have a trace backend?

Reply:
probably not, but i think we query gorilla for the stream run there, so in theory hasWeavedata could be true with no tracebackend